provisioning/miniprov/hare_mp/main.py

```diff
@@ -445,10 +445,36 @@ def start_mkfs_parallel(hostname: str, hare_config_dir: str):
                       f'{perf_result}\n\n')
 
 
+@repeat_if_fails()
+def wait_for_mkfs_to_complete_on_all_nodes(utils: Utils,
+                                           cns_utils: ConsulUtil):
+    nodes = utils.get_io_nodes()
+    for node in nodes:
+        while not cns_utils.kv.kv_get(f'mkfs_done/{node}', recurse=True):
+            pass
```

Comment:
If mkfs fails for any node, then do we need to add some failure signature in consul kv for that node, in exception handler for function start_mkfs_parallel()?

Reply:
Don't think so, in case of fialure, `start_mkfs_parallel()`  raises an exception,
```
    if not all(rc == 0 for rc in ret_values):
        raise RuntimeError('One or more mkfs processes failed to start. '
                           'Please check the logs above for details.')
```
that is handled in init(),
```
    except Exception as error:
        if hax_starter:
            hax_starter.stop()
        if consul_starter:
            consul_starter.stop()
        raise RuntimeError(f'Error while initializing cluster :key={error}')
```
`wait_for_mkfs_to_complete_on_all_nodes()` will not be executed, `hare_setup` will bail out.

Reply:
I would suggest to add a sleep with some random jitter instead of a simple `pass`. I believe such a loop can make unnecessary large load on Consul KV which may interfere overall bootstrap procedures (i.e. add delays) within the whole cluster.

Reply:
Yep, updated.